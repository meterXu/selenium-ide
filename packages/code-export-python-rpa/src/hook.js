// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the specific language governing permissions and limitations
// under the License.

import { codeExport as exporter } from '@seleniumhq/side-utils'

const emitters = {
  afterAll: empty,
  afterEach: empty,
  beforeAll,
  beforeEach: empty,
  declareDependencies,
  declareMethods: empty,
  declareVariables: empty,
  inEachBegin: inEachBegin,
  inEachEnd: inEachEnd,
  testEnd: testEnd,
}

function generate(hookName,project) {
  return new exporter.hook(emitters[hookName](project))
}

export function generateHooks(project) {
  let result = {}
  Object.keys(emitters).forEach(hookName => {
    result[hookName] = generate(hookName,project)
  })
  return result
}

function beforeAll(project) {
  let delay = project ? project.delay : 300
  let implicitlyWait = project ? project.implicitlyWait : 3000
  delay = (delay / 1000).toFixed(1)
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: 'class RPAInfo:' },
        { level: 1, statement: 'driver = None' },
        { level: 1, statement: `delay = ${delay}` },
        { level: 1, statement: `waitTime = ${implicitlyWait}` },
        { level: 1, statement: 'vars = {}' },
        { level: 0, statement: '' },
        { level: 1, statement: 'def getDriver(self):' },
        { level: 2, statement: 'time.sleep(self.delay)' },
        { level: 2, statement: 'if self.driver is None:' },
        { level: 3, statement: 'self.driver = webdriver.Chrome()' },
        { level: 2, statement: 'self.driver.implicitly_wait(self.waitTime)' },
        { level: 2, statement: 'return self.driver' },
      ],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '' }, { level: 0, statement: '' }],
    },
    registrationLevel: 1,
  }
  return params
}

function afterAll() {
  const params = {
    startingSyntax: {
      commands: [{ level: 0, statement: 'def teardown_class(cls):' }],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '' }],
    },
    registrationLevel: 1,
  }
  return params
}

function afterEach() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 1, statement: 'def quitCase(self):' },
        { level: 2, statement: 'self.driver.quit()' },
      ],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '' }],
    },
  }
  return params
}

function beforeEach() {
  const params = {
    startingSyntax: ({ browserName, gridUrl } = {}) => ({
      commands: [],
    }),
    endingSyntax: {
      commands: [{ level: 0, statement: '' }],
    },
  }
  return params
}

function inEachBegin() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 1, statement: 'self = RPAInfo()' },
        { level: 1, statement: 'self.driver = driver' },
        { level: 1, statement: 'try:' },
      ],
    },
    endingSyntax: {
      commands: [],
    },
  }
  return params
}

function inEachEnd() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 1, statement: 'except Exception as Error:' },
        { level: 2, statement: 'print(Error)' },
        { level: 1, statement: 'return self' },
        { level: 1, statement: '' },
        { level: 1, statement: '' },
      ],
    },
    endingSyntax: {
      commands: [],
    },
  }
  return params
}

function testEnd() {
  const params = {
    startingSyntax: {
      commands: [],
    },
    endingSyntax: {
      commands: [],
    },
  }
  return params
}

function declareDependencies() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: 'import pytest' },
        { level: 0, statement: 'import time' },
        { level: 0, statement: 'import json' },
        { level: 0, statement: 'from selenium import webdriver' },
        {
          level: 0,
          statement: 'from selenium.webdriver.common.by import By',
        },
        {
          level: 0,
          statement:
            'from selenium.webdriver.common.action_chains import ActionChains',
        },
        {
          level: 0,
          statement:
            'from selenium.webdriver.support import expected_conditions',
        },
        {
          level: 0,
          statement:
            'from selenium.webdriver.support.wait import WebDriverWait',
        },
        {
          level: 0,
          statement: 'from selenium.webdriver.common.keys import Keys',
        },
        {
          level: 0,
          statement:
            'from selenium.webdriver.common.desired_capabilities import DesiredCapabilities',
        },
        {
          level: 0,
          statement: '',
        },
        {
          level: 0,
          statement: '',
        },
      ],
    },
  }
  return params
}

function empty() {
  return {}
}
