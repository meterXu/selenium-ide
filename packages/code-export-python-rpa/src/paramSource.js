export default {
  sourceList: [
    {
      target: '',
      isParam: true,
      directionType: 'value',
      directionValue: 'xxxx',
    },
  ],
}
