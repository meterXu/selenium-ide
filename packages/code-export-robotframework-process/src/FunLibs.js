class FunLibs {
  constructor() {
    this.libs = {
      读取excel: {
        name: 'read_excel_to_df_withName',
        params: ['path', 'sheet'],
        return: 'dataframe',
        generate: (path, sheet) => {
          return `$\{readData\}=    read_excel_to_df_withName    ${path}    ${sheet}`
        },
      },
      转字符串: {
        name: 'evaluate',
        params: ['value'],
        return: 'resValue',
        generate: (key, index) => {
          return `$\{${key}\}    evaluate    str($\{readData.${key}[${index}]\})`
        },
      },
      创建数组: {
        name: 'Create List',
        params: ['value'],
        return: ['list'],
        generate: () => {
          return `$\{pgSourceArray\}=    Create List    $\{self.driver.page_source\}`
        },
      },
      写入excel: {
        name: 'read_excel_to_df_withName',
        params: ['path', 'sheet'],
        return: null,
        generate: (path, sheet) => {
          return `write_excel_row_fordf    ${path} ${sheet}`
        },
      },
    }
  }
}
export default new FunLibs()
