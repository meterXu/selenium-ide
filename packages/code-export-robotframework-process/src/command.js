// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.
import funlist from './FunLibs'
export default class emitters {
  constructor(process, project) {
    this.process = process
    this.projetc = project
    this.generateBody = this.generateBody.bind(this)
    this.generateHead = this.generateHead.bind(this)
  }
  generateHead(command) {
    return this.generateCode(command)
  }
  generateBody() {
    let bodycodes = this.process.graphData.map((c, i) => {
      switch (c.type) {
        case 1: {
          return this.generateCaseCode(c, i)
        }
        case 2:
          break
      }
    })
    let bodycode = bodycodes.join('\r\n')
    return bodycode
  }

  generateCaseCode(prcItem, index) {
    let sourceData = []
    if (this.projetc.sourceData) {
      sourceData = [
        ...this.projetc.sourceData.read,
        ...this.projetc.sourceData.write,
      ]
    }
    let test = this.projetc.tests.find(c => c.id === prcItem.data.caseId)
    let source = sourceData.find(c => c.id === prcItem.data.sourceId)
    let funName = undefined
    if (test) {
      funName = test.name.trim()
    }
    const ioReadCommands = this.generateIoRead(source)
    const ioWriteCommands = this.generateIoWrite(source)
    let formatCommands = []
    let paramstr = prcItem.data.paraValues
      .map(c => {
        formatCommands.push({
          level: 1,
          statement: funlist.libs.转字符串.generate(c, 0),
        })
        return `$\{${c}\}`
      })
      .join(' ')
    let firstParam = index === 0 ? '${None}' : '${self}'
    const commands = [
      {
        level: 1,
        statement: `$\{self\}=    ${funName}    ${firstParam}    ${paramstr}`,
      },
    ]
    return this.generateCode([
      ...ioReadCommands,
      ...formatCommands,
      ...commands,
      ...ioWriteCommands,
    ])
  }

  generateCode(commands) {
    let codes = commands.map(c => {
      return this.generateTab(c.level) + c.statement
    })
    return codes.join('\n')
  }
  generateTab(level) {
    let tabs = Array(level).fill('    ')
    return tabs.join('')
  }

  generateIoRead(source) {
    if (source && source.type === 0) {
      switch (source.data.type) {
        case 0: {
          return [
            {
              level: 1,
              statement: funlist.libs.读取excel.generate(
                source.data.pro.path,
                source.data.pro.sheet
              ),
            },
          ]
        }
        default:
          return []
      }
    }
    return []
  }

  generateIoWrite(source) {
    if (source && source.type === 1) {
      switch (source.data.type) {
        case 0: {
          return [
            {
              level: 1,
              statement: funlist.libs.写入excel.generate(
                source.data.pro.path,
                source.data.pro.sheet
              ),
            },
          ]
        }
        default:
          return []
      }
    }
    return []
  }
}
