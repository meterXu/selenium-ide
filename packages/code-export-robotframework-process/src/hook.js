// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the specific language governing permissions and limitations
// under the License.

import { codeExport as exporter } from '@seleniumhq/side-utils'

const emitters = {
  afterAll: empty,
  afterEach: empty,
  beforeAll,
  beforeEach: empty,
  declareDependencies,
  declareMethods: empty,
  declareVariables: empty,
  inEachBegin: inEachBegin,
  inEachEnd: inEachEnd,
  testEnd: testEnd,
}

function generate(hookName, project) {
  return new exporter.hook(emitters[hookName](project))
}

export function generateHooks(project) {
  let result = {}
  Object.keys(emitters).forEach(hookName => {
    result[hookName] = generate(hookName, project)
  })
  return result
}

function beforeAll(project) {
  const params = {
    startingSyntax: {
      commands: [

      ],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '' }, { level: 0, statement: '' }],
    },
    registrationLevel: 1,
  }
  return params
}

function afterAll() {
  const params = {
    startingSyntax: {
      commands: [{ level: 0, statement: 'def teardown_class(cls):' }],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '' }],
    },
    registrationLevel: 1,
  }
  return params
}

function afterEach() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 1, statement: 'def quitCase(self):' },
        { level: 2, statement: 'self.driver.quit()' },
      ],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '' }],
    },
  }
  return params
}

function beforeEach() {
  const params = {
    startingSyntax: ({ browserName, gridUrl } = {}) => ({
      commands: [],
    }),
    endingSyntax: {
      commands: [{ level: 0, statement: '' }],
    },
  }
  return params
}

function inEachBegin() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 1, statement: 'self = RPAInfo()' },
        { level: 1, statement: 'self.driver = driver' },
      ],
    },
    endingSyntax: {
      commands: [],
    },
  }
  return params
}

function inEachEnd() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 1, statement: 'return self' },
        { level: 1, statement: '' },
        { level: 1, statement: '' },
      ],
    },
    endingSyntax: {
      commands: [],
    },
  }
  return params
}

function testEnd() {
  const params = {
    startingSyntax: {
      commands: [],
    },
    endingSyntax: {
      commands: [],
    },
  }
  return params
}

function declareDependencies() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: '*** Settings ***' },
        { level: 0, statement: 'Library           JetExcelClient' },
        { level: 0, statement: `Library           JetOracleClient` },
        { level: 0, statement: `Library           JetHtmlAnalyze` },
        { level: 0, statement: `` },
        { level: 0, statement: '*** Test Cases ***' },
      ],
    },
  }
  return params
}

function empty() {
  return {}
}
