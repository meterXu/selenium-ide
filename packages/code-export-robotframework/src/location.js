// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import { codeExport as exporter } from '@seleniumhq/side-utils'

const emitters = {
  id: emitId,
  name: emitName,
  link: emitLink,
  linkText: emitLink,
  partialLinkText: emitPartialLinkText,
  css: emitCss,
  xpath: emitXpath,
  jquery: emitJQuery,
  class: emitClass,
  identifier: emitIdentifier,
  tag: emitTag,
  dom: emitDom,
  sizzle: emitSizzle,
  default: emitDefault,
}

export function emit(location) {
  return exporter.emit.location(location, emitters)
}

export default {
  emit,
}

function emitId(selector) {
  return Promise.resolve(`id:${selector}`)
}

function emitName(selector) {
  return Promise.resolve(`name:${selector}`)
}

function emitLink(selector) {
  return Promise.resolve(`link:${selector}`)
}

function emitPartialLinkText(selector) {
  return Promise.resolve(`partial link:${selector}`)
}

function emitCss(selector) {
  return Promise.resolve(`css:${selector}`)
}

function emitXpath(selector) {
  return Promise.resolve(`xpath:${selector}`)
}

function emitJQuery(selector) {
  return Promise.resolve(`jquery:${selector}`)
}

function emitClass(selector) {
  return Promise.resolve(`class:${selector}`)
}

function emitIdentifier(selector) {
  return Promise.resolve(`identifier:${selector}`)
}

function emitTag(selector) {
  return Promise.resolve(`tag:${selector}`)
}

function emitDom(selector) {
  return Promise.resolve(`dom:${selector}`)
}

function emitSizzle(selector) {
  return Promise.resolve(`sizzle:${selector}`)
}

function emitDefault(selector) {
  return Promise.resolve(`default:${selector}`)
}
