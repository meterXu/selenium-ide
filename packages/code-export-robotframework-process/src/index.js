// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import { codeExport as exporter } from '../../side-utils/dist'
import emitter from './command'
import location from './location'
import { generateHooks } from './hook'

// Define language options
export const displayName = 'RobotFramework process'

export let opts = {}
opts.hooks = generateHooks()
opts.fileExtension = '.robot'
opts.commandPrefixPadding = '  '
opts.terminatingKeyword = ''
opts.commentPrefix = '#'
opts.testLevel = '0'
opts.commandLevel = '0'
opts.generateMethodDeclaration = generateMethodDeclaration
// Create generators for dynamic string creation of primary entities (e.g., filename, methods, test, and suite)
function generateTestDeclaration(process) {
  return `${process.name}`
}
function generateMethodDeclaration(name) {
  return `def ${exporter.parsers.uncapitalize(
    exporter.parsers.sanitizeName(name)
  )}(self):`
}
// eslint-disable-next-line no-unused-vars
function generateSuiteDeclaration(name) {
  return ``
}

function generateFilename(name) {
  return `process_${exporter.parsers.uncapitalize(
    exporter.parsers.sanitizeName(name)
  )}${opts.fileExtension}`
}

// Emit an individual test, wrapped in a suite (using the test name as the suite name)
export async function emitProcess({
  baseUrl,
  test,
  tests,
  process,
  project,
  enableOriginTracing,
  beforeEachOptions,
}) {
  opts.hooks = generateHooks(project)
  global.baseUrl = baseUrl
  const testDeclaration = generateTestDeclaration(process)
  let body = new emitter(process, project).generateBody()
  return {
    filename: `${process.name}.robot`,
    body: `${testDeclaration}\r\n${body}`,
  }
}

export default {
  emit: {
    test: emitProcess,
    locator: location.emit,
  },
  register: {
    command: emitter.register,
    variable: opts.hooks.declareVariables.register,
    dependency: opts.hooks.declareDependencies.register,
    beforeAll: opts.hooks.beforeAll.register,
    beforeEach: opts.hooks.beforeEach.register,
    afterEach: opts.hooks.afterEach.register,
    afterAll: opts.hooks.afterAll.register,
    inEachBegin: opts.hooks.inEachBegin.register,
    inEachEnd: opts.hooks.inEachEnd.register,
  },
}
