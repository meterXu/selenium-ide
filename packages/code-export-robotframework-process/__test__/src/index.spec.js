// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import fs from 'fs'
import path from 'path'
import { emitProcess } from '../../src'

function readFile(filename) {
  return JSON.parse(
    fs.readFileSync(
      path.join(
        __dirname,
        '..',
        '..',
        '..',
        'side-utils',
        '__test__',
        'test-files',
        filename
      )
    )
  )
}

describe('Code Export Python pytest', () => {
  it('should export a test', async () => {
    const project = readFile('single-test.side')
    const results = await emitProcess({
      baseUrl: project.url,
      test: project.tests[0],
      tests: project.tests,
      process: project.processData[0],
      project: project,
    })
    console.log(results.body)
    expect(results.body).toBeDefined()
    expect(results.body).toMatchSnapshot()
  })
})
