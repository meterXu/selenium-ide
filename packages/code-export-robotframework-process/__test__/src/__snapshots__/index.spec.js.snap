// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Export Python pytest should export a suite 1`] = `
"# Generated by JetRecord
import pytest
import time
import json
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities


class RPAInfo:
  driver = None
  delay = 0.3
  waitTime = 2100
  vars = {}

  def getDriver(self):
    time.sleep(self.delay)
    if self.driver is None:
      self.driver = webdriver.Chrome()
    self.driver.implicitly_wait(self.waitTime)
    return self.driver


def valid credentials(driver, parm1, parm2):
  self = RPAInfo()
  self.driver = driver
  self.getDriver().get(\\"http://the-internet.herokuapp.com/login\\")
  self.getDriver().set_window_size(1440, 1177)
  self.getDriver().find_element(By.ID, \\"username\\").clear()
  self.getDriver().find_element(By.ID, \\"username\\").send_keys(parm1)
  self.getDriver().find_element(By.ID, \\"password\\").clear()
  self.getDriver().find_element(By.ID, \\"password\\").send_keys(parm2)
  self.getDriver().find_element(By.CSS_SELECTOR, \\"#login button\\").click()
  assert self.getDriver().find_element(By.CSS_SELECTOR, \\".flash.success\\").text == \\"You logged into a secure area!\\\\\\\\\\\\\\\\n×\\"
  return self
  
  
def invalid credentials(driver):
  self = RPAInfo()
  self.driver = driver
  self.getDriver().get(\\"http://the-internet.herokuapp.com/login\\")
  self.getDriver().set_window_size(1440, 1177)
  self.getDriver().find_element(By.ID, \\"username\\").clear()
  self.getDriver().find_element(By.ID, \\"username\\").send_keys(\\"blah\\")
  self.getDriver().find_element(By.ID, \\"password\\").clear()
  self.getDriver().find_element(By.ID, \\"password\\").send_keys(\\"blah\\")
  self.getDriver().find_element(By.CSS_SELECTOR, \\"#login button\\").click()
  assert self.getDriver().find_element(By.ID, \\"flash\\").text == \\"Your username is invalid!\\\\\\\\\\\\\\\\n×\\"
  return self
  
  
"
`;

exports[`Code Export Python pytest should export a test 1`] = `
"# Generated by JetRecord
import pytest
import time
import json
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities


class RPAInfo:
  driver = None
  delay = 0.3
  waitTime = 1400
  vars = {}

  def getDriver(self):
    time.sleep(self.delay)
    if self.driver is None:
      self.driver = webdriver.Chrome()
    self.driver.implicitly_wait(self.waitTime)
    return self.driver


def login(driver, parm1, parm2, parm3):
  self = RPAInfo()
  self.driver = driver
  self.getDriver().get(\\"http://the-internet.herokuapp.com/login\\")
  self.getDriver().set_window_size(1440, 1177)
  self.getDriver().find_element(By.XPATH, \\"//input[@id=\\\\'\\"+parm1+\\"\\\\']\\").click()
  self.getDriver().find_element(By.ID, \\"username\\").clear()
  self.getDriver().find_element(By.ID, \\"username\\").send_keys(parm2)
  self.getDriver().find_element(By.ID, \\"password\\").clear()
  self.getDriver().find_element(By.ID, \\"password\\").send_keys(parm3)
  self.getDriver().find_element(By.CSS_SELECTOR, \\"#login button\\").click()
  assert self.getDriver().find_element(By.CSS_SELECTOR, \\".flash.success\\").text == \\"You logged into a secure area!\\\\\\\\\\\\\\\\n×\\"
  return self
  
  
"
`;
